### PR TITLE
Add way to configure the dnf/yum plugin

### PR DESCRIPTION
**Pull request by jcpunk**
#### Pull Request (PR) description
With EL8+ the `dnf` plugin has gained a few features that are nice to set.  This adds a parameter to allow folks to configure those settings easily.

I'll confess I don't like the name I ended up with very much...

### CODE REVIEW

**Comment by jcpunk** [comment]
I don't think these test errors are something I can fix....

**Comment by bastelfreak** [comment]
I merged https://github.com/voxpupuli/puppet-rhsm/pull/118, please rebase

**Comment by jcpunk** [comment]
Rebased, I'm not sure why the checks aren't doing what they should.....

File: spec/classes/init_spec.rb [hunk 1]
```diff
@@ -105,6 +119,33 @@
...
+      context 'with specific plugin options' do
+        let :params do
+          {
+            rh_password: 'password',
+            rh_user: 'username',
+            plugin_settings: { 'main' => { 'enabled' => 0, 'disable_system_repos' => 1 } }
+          }
+        end
+
+        if facts[:os]['family'] == 'RedHat' && facts[:os]['release']['major'] == 7
```
Comment (root-expert) [inline spec/classes/init_spec.rb:132]: `facts[:os]['release']['major']` is a string, you should either convert it to integer with `to_i` or use `versioncmp` to compare the to versions.

Also RedHat 6 falls in the else branch which according to the puppet code is wrong, that's why redhat 6 are failing.

**Comment by bastelfreak** [comment]
thanks for the work!

**Pull request merged**